website/blog/how-to-migrate-an-ios-app-to-bazel.md

```diff
@@ -0,0 +1,1470 @@
+
+If `srcs` is empty,
+a `cc_library` target acts a collection of
+headers,
+defines,
+include paths,
+or linkopts,
+which are propagated to dependent targets.
+
+:::
+
+[cc_library]: https://bazel.build/reference/be/c-cpp#cc_library
+
+#### `experimental_mixed_language_library`
+
+An [`experimental_mixed_language_library`][experimental_mixed_language_library] target defines an Objective-C and Swift mixed-language static library.
+Use it for compiling mixed-language modules.
+
+`experimental_mixed_language_library` is actually a [macro][macro] that creates
+a [`swift_library`](#swift_library),
+an [`objc_library`](#objc_library),
+and some [modulemaps][modulemaps] to tie them together.
+
+:::note
+
+Due to poor build performance,
+it is not recommended to have mixed-language modules.
+It's recommended to only use this macro as a migration stopgap until you are able to demix them.
+
+:::
+
+[experimental_mixed_language_library]: https://github.com/bazelbuild/rules_apple/blob/3.3.0/doc/rules-apple.md#experimental_mixed_language_library
+[macro]: https://bazel.build/versions/7.0.0/extending/macros
+[modulemaps]: https://clang.llvm.org/docs/Modules.html#module-maps
+
+#### `apple_intent_library`
+
+An [`apple_intent_library`][apple_intent_library] target generates source files for an `.intentdefinition` file.
+Use it if you have `.intentdefinition` resources.
+
+:::note
+
+The [`swift_intent_library`][swift_intent_library] and [`objc_intent_library`][objc_intent_library] macros wrap `apple_intent_library` with [`swift_library`](#swift_library) and [`objc_library`](#objc_library) targets respectively.
+Use them instead of `apple_intent_library` directly if possible.
+
+[objc_intent_library]: https://github.com/bazelbuild/rules_apple/blob/3.3.0/doc/rules-resources.md#objc_intent_library
+[swift_intent_library]: https://github.com/bazelbuild/rules_apple/blob/3.3.0/doc/rules-resources.md#swift_intent_library
+
+:::
+
+[apple_intent_library]: https://github.com/bazelbuild/rules_apple/blob/3.3.0/doc/rules-resources.md#apple_intent_library
+
+#### `apple_resource_bundle`
+
+An [`apple_resource_bundle`][apple_resource_bundle] target generates a resource bundle.
+Use it if you require certain resources to be placed in a named `.bundle`,
+instead of directly placed in your top level bundle
+(e.g. an `.app`, `.framework` , or `.xctest`).
+
+:::note
+
+`apple_resource_bundle` targets need to be listed in `data` attributes,
+not `deps` attributes.
+
+:::
+
+[apple_resource_bundle]: https://github.com/bazelbuild/rules_apple/blob/3.3.0/doc/rules-resources.md#apple_resource_bundle
+
+#### `ios_application`
+
+An [`ios_application`][ios_application] target generates an application bundle.
+
+Unlike Xcode,
+Bazel separates compilation and bundling targets.
+This means that an `ios_application` target
+(which is a bundling target)
+doesn't list its source files,
+and instead requires that its primary module be its single [static library](#swift_library) dependency
```

Comment:
Perhaps a little odd to link `static library` to `swift_library` section here. Might be more clear to say:
```suggestion
and instead requires that its primary module be a single static library dependency, such as a [swift_library](#swift_library) or an [objc_library](#objc_library) (which is a compilation target).
```